dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java

```diff
@@ -1261,927 +1290,2252 @@ private static void addSummaryMessages(final int headerCount, final int importab
-            HashSet<String> keyContentUpdated, List<Permission> contentTypePermissions, List<UniqueFieldBean> uniqueFieldBeans, List<Field> uniqueFields, HashMap<Integer, Relationship> relationships, HashMap<Integer, Boolean> onlyChild, HashMap<Integer, Boolean> onlyParent,
-            boolean sameKeyBatchInsert, String wfActionId, final HttpServletRequest request ) throws DotRuntimeException {
+     * Processes a single line from the CSV import file. This method handles all aspects
+     * of importing a content line including:
+     * <ul>
+     *   <li>Field validation and processing</li>
+     *   <li>Content matching and updates</li>
+     *   <li>Relationship processing</li>
+     *   <li>Workflow execution</li>
+     *   <li>Category handling</li>
+     * </ul>
+     *
+     * @param line CSV line data as string array
+     * @param currentHostId ID of the current host/site
+     * @param contentType Content type structure definition
+     * @param preview If true, validates without saving changes
+     * @param isMultilingual If true, handles multilingual content
+     * @param user User performing the import
+     * @param identifier Optional identifier for content updates
+     * @param wfActionIdIndex Index of workflow action ID in CSV, -1 if none
+     * @param lineNumber Current line number in CSV file
+     * @param language Language ID for the content
+     * @param headers Map of column indices to field definitions
+     * @param keyFields Map of key fields used for content matching
+     * @param choosenKeyField Buffer tracking chosen key fields
+     * @param keyContentUpdated Set tracking updated content keys
+     * @param contentTypePermissions Content type permissions
+     * @param uniqueFieldBeans List tracking unique field values
+     * @param uniqueFields List of fields marked as unique
+     * @param relationships Map of relationship definitions
+     * @param onlyChild Map tracking child-only relationships
+     * @param onlyParent Map tracking parent-only relationships
+     * @param sameKeyBatchInsert True if batch contains multiple rows with same key
+     * @param wfActionId Workflow action ID to execute
+     * @param request HTTP request context
+     * @return Line import results including validation messages and processing outcomes
+     * @throws DotRuntimeException If a critical error occurs during import
+     */
+    private static LineImportResult importLine(
+            final String[] line,
+            final String currentHostId,
+            final Structure contentType,
+            final boolean preview,
+            boolean isMultilingual,
+            final User user,
+            final String identifier,
+            final int wfActionIdIndex,
+            final int lineNumber,
+            final long language,
+            final HashMap<Integer, Field> headers,
+            final HashMap<Integer, Field> keyFields,
+            final StringBuffer choosenKeyField,
```

Comment:
Cant we replace this StringBuffer with StringBuilder?

Reply:
Done